starlette/middleware/base.py

```diff
@@ -198,20 +198,33 @@ async def dispatch(
         raise NotImplementedError()  # pragma: no cover
 
 
-class _StreamingResponse(StreamingResponse):
+class _StreamingResponse(Response):
     def __init__(
         self,
-        content: ContentStream,
+        content: AsyncContentStream,
         status_code: int = 200,
         headers: typing.Mapping[str, str] | None = None,
         media_type: str | None = None,
-        background: BackgroundTask | None = None,
```

Comment:
is the removal of `background` here intentional? We use `background` for both actual background tasks but also to detect when a streaming request was cancelled by the client (`request.is_disconnected` had some minor overhead).

Reply:
Cutting a PR for this

Reply:
https://github.com/encode/starlette/pull/2687